### PR TITLE
Skeleton scaler 

### PR DESCRIPTION
**Pull request by CloudViking**
#7 - Skeleton Scaler 

- Created `/dev/scaler/function.psm1` as an example template for future scalers
- Major documentation updates covering Scaler Module Logic
    - Logical process to create new scalers
    - Mapping Graph API data to PowerShell commands 
- Documentation updates to cover custom tag values and tagMap data

### CODE REVIEW

File: dev/scaler/function.psm1 [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    }
+
+    $config = @{ }
+    $tags = $tagData.tags
+
+#######################################################################################################################
+##  
+## SCALING UP
+##
+## You will need to switch off of the direction sent from the Engine function.  Scaling up is the easier
```
Comment (NillsF) [inline dev/scaler/function.psm1:84]: I don't understand the first sentence.

Comment (NillsF) [reply dev/scaler/function.psm1:84]: I assume you want to say: "Scaling up means bringing the resource back to its original state."

Comment (DCMattyG) [reply dev/scaler/function.psm1:84]: I believe he is referencing the "Switch" statement in PowerShell. I agree the 'phrasiology' needs some work, haha.

File: docs/scaler/README.md [hunk 2]
```diff
@@ -2,13 +2,20 @@
...
+    - **Tag Map Data**
+        - This is the map data to inform Bellhop which tags to use for scale operations
+        - _All Tag Keys are customizable now from the "Advanced" tab when using the "Deploy to Azure" button_
+    - **Azure Resource Graph Results**
+        - Results of the Azure Resource Graph API query for the target resource
+- Entire resize operation is executed by the Scaler module
 
 ## Scaler Module Details
-Instructions for creating new modules, as well as detailed information around the each current supported Azure Resource can be found in the [Scaler Modules](/scaler/modules/README.md) section of this documentation.
+Instructions for creating new modules, as well as detailed information around the each currently supported Azure Resource can be found in the [Scaler Modules](/scaler/modules/README.md) section of the documentation.
```
Comment (NillsF) [inline docs/scaler/README.md:21]: Do we call it "new modules" or "new scalers"?

**Review by NillsF — APPROVED** [review]
Actually looks really good to me. Great work team.

File: docs/scaler/README.md (hunk 2 above)
Comment (DCMattyG) [reply docs/scaler/README.md:21]: I believe the terminology we're using is "Scaler" for the actual Function that is running, and "Scaler modules" for the resource specific modules within the Scaler function.

File: dev/scaler/function.psm1 (hunk 1 above)
Comment (NillsF) [reply dev/scaler/function.psm1:84]: Ah yes, now it makes sense. Why not call it: The scaling starts with a switch-statement based on the scale direction (either up/down)

**Review by NillsF — APPROVED** [review]
look good

**Review by DCMattyG — APPROVED** [review]
So much good documentation, great work Tyler!

**Pull request merged**